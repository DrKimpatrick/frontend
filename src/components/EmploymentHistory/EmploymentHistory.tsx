import React, { FC, Fragment, useEffect } from 'react';
import { withRouter } from 'react-router-dom';
import './EmploymentHistory.scss';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import ArrowRightAltTwoToneIcon from '@material-ui/icons/ArrowRightAltTwoTone';
import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';
import NavBar from 'components/Layout/NavBar/NavBar';

type props = {};
const EmploymentHistory: FC<props> = (props: any) => {

  useEffect(() => {
    const token = localStorage.getItem('token');
    if (!token) {
      props.history.push('/login');
    }
  });

  return (
    <Fragment>
      <NavBar />
      <section className="employment-history-section w-1/3 m-auto text-textGray">
        <div className="flex relative h-auto my-8">
          <div
            className="back-arrow cursor-pointer"
            onClick={() => props.history.push('/recent-employer')}
          >
            <ArrowBackTwoToneIcon />
          </div>
          <h1 className="font-bold text-xl title">Employment History</h1>
        </div>

        <div className="text-textGray mt-8 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Company
          </div>
          <div className="py-3 px-4 card-content">Google</div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Supervisor
          </div>
          <div className="py-3 px-4 card-content">Mr. Black Pather</div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Duration
          </div>
          <div className="py-3 px-4 card-content">Jan 8, 2019 - Current</div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Responsibilities
          </div>
          <div className="py-3 px-4 card-content">
            Lorem ipsum must be a head or manager lorem ipsum
          </div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Accomplishments
          </div>
          <div className="py-3 px-4 card-content">
            Lorem ipsum must be a head or manager lorem ipsum
          </div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Favorite Project
          </div>
          <div className="py-3 px-4 card-content">TTLCC, TTLCC</div>
        </div>
        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Skills Used
          </div>
          <div className="py-3 px-4 card-content h-32"></div>
        </div>

        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="add-employer-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-64 rounded-sm shadow flex justify-around"
          >
            <span className="">Add another employer</span>
            <AddCircleOutlineOutlinedIcon />
          </button>
        </div>
        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
            onClick={() => props.history.push('/employment-history-list')}
          >
            <span className="">Next</span> <ArrowRightAltTwoToneIcon />
          </button>
        </div>
      </section>
    </Fragment>
  );
};

export default withRouter(EmploymentHistory);
