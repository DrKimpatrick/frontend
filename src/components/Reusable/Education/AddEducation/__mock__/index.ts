export const initialState = {
  users: {
    currentUser: {
      isLoggedIn: true,
      loading: false,
      error: null,
      message: null,
      created: false,
      data: {
        firstName: 'John',
        lastName: 'Smith',
        profilePicture:
          'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
      }
    },
    user: {
      _id: 'id',
      username: 'username',
      email: 'email',
      roles: ['talent'],
      updatedAt: '2020-10-10',
      dateRegistered: '2020-10-10',
      paymentStatus: 'verified',
      featureChoice: 'local',
      verified: true
    }
  },
  messages: {
    message: 'sent'
  },
  educations: {
    loading: false,
    errors: {
      errors: null
    },
    education: {
      _id: 'id',
      schoolName: 'school name',
      level: 'level',
      degreeOrCertification: 'education',
      specializations: ['specialization'],
      startDate: '2020-10-10',
      endDate: '2020-10-5',
      accomplishments: ['snow'],
      isCurrentEducation: false,
      verificationStatus: 'verified'
    }
  }
};
