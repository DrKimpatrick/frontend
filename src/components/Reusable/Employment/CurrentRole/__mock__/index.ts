export const initialState = {
  users: {
    currentUser: {
      isLoggedIn: true,
      loading: false,
      error: null,
      message: null,
      created: false,
      data: {
        firstName: 'John',
        lastName: 'Smith',
        profilePicture:
          'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
      }
    },
    user: {
      _id: 'id',
      username: 'username',
      email: 'email',
      roles: ['talent'],
      updatedAt: '2020-10-10',
      dateRegistered: '2020-10-10',
      paymentStatus: 'verified',
      featureChoice: 'local',
      verified: true
    }
  },
  employments: {
    employment: {
      companyName: 'snow'
    },
    errors: null,
    loading: false
  },
  messages: {
    message: 'sent'
  },
  educations: {
    education: {
      schoolName: 'schoolName'
    }
  }
};
