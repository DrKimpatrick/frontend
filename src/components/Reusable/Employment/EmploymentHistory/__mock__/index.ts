export const initialState = {
  users: {
    currentUser: {
      isLoggedIn: true,
      loading: false,
      error: null,
      message: null,
      created: false,
      data: {
        firstName: 'John',
        lastName: 'Smith',
        profilePicture:
          'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
      }
    },
    user: {
      _id: 'id',
      username: 'username',
      email: 'email',
      roles: ['talent'],
      updatedAt: '2020-10-10',
      dateRegistered: '2020-10-10',
      paymentStatus: 'verified',
      featureChoice: 'local',
      verified: true
    }
  },
  employments: {
    employment: {
      companyName: 'snow',
      supervisor: 'supervisor',
      title: 'title',
      startDate: '2020-10-5',
      responsibilities: ['res'],
      accomplishments: ['res'],
      favoriteProject: 'res',
      skillsUsed: []
    },
    errors: null,
    loading: false
  },
  messages: {
    message: 'sent'
  },
  educations: {
    education: {
      schoolName: 'schoolName'
    }
  },
  skills: {
    userSkill: [
      {
        supervisor: {
          detail: {
            name: 'user',
            email: 'email@gmail.com',
            phoneNumber: '0786601003'
          },
          name: 'HR'
        },
        skillsUsed: [],
        responsibilities: [],
        accomplishments: [],
        verificationStatus: 'unverified',
        isCurrentPosition: true,
        _id: '5fcf65a2c4c797a20acc1c99',
        title: 'Product Manager',
        startDate: '2020-10-15T00:00:00.000Z',
        endDate: null,
        companyName: 'heza',
        favoriteProject: 'tembea',
        userId: '5fcdf47ea1d1dd1564a21613',
        createdAt: '2020-12-08T11:38:10.729Z',
        updatedAt: '2020-12-08T11:38:10.729Z'
      }
    ]
  }
};
