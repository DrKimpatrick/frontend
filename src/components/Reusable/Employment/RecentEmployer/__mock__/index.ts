export const initialState = {
  users: {
    currentUser: {
      isLoggedIn: true,
      loading: false,
      error: null,
      message: null,
      created: false,
      data: {
        firstName: 'John',
        lastName: 'Smith',
        profilePicture:
          'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
      }
    },
    user: {
      _id: 'id',
      username: 'username',
      email: 'email',
      roles: ['talent'],
      updatedAt: '2020-10-10',
      dateRegistered: '2020-10-10',
      paymentStatus: 'verified',
      featureChoice: 'local',
      verified: true
    }
  },
  employments: {
    employment: {
      companyName: 'snow',
      supervisor: 'supervisor',
      title: 'title',
      startDate: '2020-10-5',
      responsibilities: ['res'],
      accomplishments: ['res'],
      favoriteProject: 'res',
      _id: 'id'
    },
    errors: null,
    loading: false,
    submitLoading: false
  },
  messages: {
    message: 'sent'
  },
  educations: {
    education: {
      schoolName: 'schoolName'
    }
  },
  skills: {
    userSkill: [
      {
        skill: {
          _id: '5fcf1c5b11c16c41f7c6c723',
          skill: 'Python',
          createdAt: '2020-12-08T06:25:31.128Z',
          updatedAt: '2020-12-08T06:25:31.128Z'
        },
        level: 'intermediate',
        verificationStatus: 'verified',
        _id: '5fcf52105cd56b9843252696',
        createdAt: '2020-12-08T10:14:40.231Z',
        updatedAt: '2020-12-08T13:05:46.873Z'
      }
    ]
  }
};
