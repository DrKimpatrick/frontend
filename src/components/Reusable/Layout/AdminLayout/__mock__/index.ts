export const initialState = {
  users: {
    currentUser: {
      isLoggedIn: true,
      loading: false,
      error: null,
      message: null,
      created: false,
      data: {
        firstName: 'John',
        lastName: 'Smith',
        profilePicture:
          'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
      }
    },
    user: {}
  }
};
