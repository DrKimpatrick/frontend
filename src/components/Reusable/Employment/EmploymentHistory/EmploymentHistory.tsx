import React, { FC, useEffect, useState, useMemo } from 'react';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import ArrowRightAltTwoToneIcon from '@material-ui/icons/ArrowRightAltTwoTone';
import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';
import { listEmployments } from 'redux/actions/employment';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from 'redux/store';
import './EmploymentHistory.scss';
import { TalentProcess } from 'redux/action-types/user';
import { Employment } from 'redux/action-types/employment';
import { setProfileProcess } from 'redux/actions/user';
import { SideLoading } from 'components/Reusable';
import { listUserSkill } from 'redux/actions/skill';
import { map, get } from 'lodash';

interface Props {
  setPreviousStep: (value: string) => void;
}

const EmploymentHistory: FC<Props> = props => {
  const [employment, setEmployment] = useState<Employment>();

  const dispatch = useDispatch();

  const { setPreviousStep } = props;

  const reducer = useSelector((state: RootState) => {
    const {
      loading,
      errors,
      employments,
      employment: singleEmployment
    } = state.employments;
    const { message } = state.messages;
    const { user } = state.users;
    const { userSkill } = state.skills;
    return {
      message,
      loading,
      errors,
      employments,
      user,
      employment: singleEmployment,
      userSkill
    };
  });

  const { employments, user, loading, userSkill } = reducer;

  useEffect(() => {
    if (reducer.employment) {
      setEmployment(reducer.employment);
    } else {
      listEmployments()(dispatch);
    }
  }, [dispatch, reducer.employment]);

  useEffect(() => {
    if (employments && employments.length > 0) {
      setEmployment(employments[0]);
    }
  }, [employments]);

  useEffect(() => {
    listUserSkill()(dispatch);
  }, [dispatch]);

  const skills = useMemo(() => {
    if (
      userSkill &&
      userSkill.length > 0 &&
      employment &&
      employment.skillsUsed &&
      employment.skillsUsed.length > 0
    ) {
      const data: any = [];

      map(userSkill, item => {
        map(employment.skillsUsed, skill => {
          if (item._id === skill._id) {
            data.push(item.skill.skill);
          }
        });
      });
      if (data.length > 0) {
        return (
          <>
            <div className="text-gray-texts mt-4 border border-gray-borders card">
              <div className=" text-white font-bold py-3 px-4 card-title">
                Skills Used
              </div>
              <div className="py-3 px-4 card-content h-32">
                {data.map((skill: any, index: number) => (
                  <p key={index}>- {skill}</p>
                ))}
              </div>
            </div>
          </>
        );
      }
      return <></>;
    }
    return <></>;
  }, [userSkill, employment]);

  if (loading) {
    return (
      <div style={{ marginTop: '200px' }}>
        <SideLoading />
      </div>
    );
  }

  if (!employment) {
    return null;
  }

  return (
    <>
      <section className="employment-history-section w-1/3 m-auto text-gray-texts">
        <div className="flex relative h-auto my-8">
          <div
            className="back-arrow cursor-pointer"
            onClick={() => setPreviousStep(TalentProcess.RecentEmployer)}
          >
            <ArrowBackTwoToneIcon />
          </div>
          <h1 className="font-bold text-xl title">Employment History Review</h1>
        </div>

        <div className="text-gray-texts mt-8 border border-gray-borders card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Company
          </div>
          <div className="py-3 px-4 card-content">{employment.companyName}</div>
        </div>

        <div className="text-gray-texts mt-4 border border-gray-borders card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Supervisor
          </div>
          <div className="py-3 px-4 card-content">
            {get(employment.supervisor, 'name', '')}
          </div>
        </div>

        <div className="text-gray-texts mt-4 border border-gray-borders card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Duration
          </div>
          <div className="py-3 px-4 card-content">
            {new Date(employment.startDate).toDateString()} to{' '}
            {employment.endDate && new Date(employment.endDate).toDateString()}
          </div>
        </div>

        <div className="text-gray-texts mt-4 border border-gray-borders card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Responsibilities
          </div>
          <div className="py-3 px-4 card-content">
            {employment.responsibilities &&
              employment.responsibilities.length > 0 &&
              employment.responsibilities.map((resp, index) => (
                <p key={index}>- {resp}</p>
              ))}
          </div>
        </div>

        {employment.accomplishments && employment.accomplishments.length > 0 && (
          <div className="text-gray-texts mt-4 border border-gray-borders card">
            <div className=" text-white font-bold py-3 px-4 card-title">
              Accomplishments
            </div>
            <div className="py-3 px-4 card-content">
              {employment.accomplishments.map((acc, index) => (
                <p key={index}>- {acc}</p>
              ))}
            </div>
          </div>
        )}

        {employment.favoriteProject && (
          <div className="text-gray-texts mt-4 border border-gray-borders card">
            <div className=" text-white font-bold py-3 px-4 card-title">
              Favorite Project
            </div>
            <div className="py-3 px-4 card-content">
              {employment.favoriteProject}
            </div>
          </div>
        )}
        {skills}
        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="add-employer-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-64 rounded-sm shadow flex justify-around"
            onClick={() => setPreviousStep(TalentProcess.RecentEmployer)}
            type="button"
          >
            <span className="">Add another employer</span>
            <AddCircleOutlineOutlinedIcon />
          </button>
        </div>
        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
            onClick={() => {
              if (user) {
                setProfileProcess({
                  userId: user._id,
                  profileProcess: TalentProcess.ListEmployment
                })(dispatch);
              }
            }}
            type="button"
          >
            <span className="">Next</span> <ArrowRightAltTwoToneIcon />
          </button>
        </div>
      </section>
    </>
  );
};

export default EmploymentHistory;
