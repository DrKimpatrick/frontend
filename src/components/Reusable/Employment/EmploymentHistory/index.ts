import EmploymentHistory from './EmploymentHistory';

export { EmploymentHistory };
