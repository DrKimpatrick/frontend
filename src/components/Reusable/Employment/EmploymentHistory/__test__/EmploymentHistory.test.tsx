import { cleanup } from '@testing-library/react';
import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import EmploymentHistory from '../EmploymentHistory';
import { initialState } from '../__mock__';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

const mockSetPreviousStep = jest.fn();

let store: any;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

describe('`EmploymentHistory` component', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <Provider store={store}>
        <Router>
          <EmploymentHistory setPreviousStep={mockSetPreviousStep} />
        </Router>
      </Provider>,
      div
    );
  });

  it('should create snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <EmploymentHistory setPreviousStep={mockSetPreviousStep} />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
