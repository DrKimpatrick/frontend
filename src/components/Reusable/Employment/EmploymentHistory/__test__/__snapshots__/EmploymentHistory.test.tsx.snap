// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`EmploymentHistory\` component should create snapshot 1`] = `
<section
  className="employment-history-section w-1/3 m-auto text-gray-texts"
>
  <div
    className="flex relative h-auto my-8"
  >
    <div
      className="back-arrow cursor-pointer"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </div>
    <h1
      className="font-bold text-xl title"
    >
      Employment History Review
    </h1>
  </div>
  <div
    className="text-gray-texts mt-8 border border-gray-borders card"
  >
    <div
      className=" text-white font-bold py-3 px-4 card-title"
    >
      Company
    </div>
    <div
      className="py-3 px-4 card-content"
    >
      snow
    </div>
  </div>
  <div
    className="text-gray-texts mt-4 border border-gray-borders card"
  >
    <div
      className=" text-white font-bold py-3 px-4 card-title"
    >
      Supervisor
    </div>
    <div
      className="py-3 px-4 card-content"
    />
  </div>
  <div
    className="text-gray-texts mt-4 border border-gray-borders card"
  >
    <div
      className=" text-white font-bold py-3 px-4 card-title"
    >
      Duration
    </div>
    <div
      className="py-3 px-4 card-content"
    >
      Mon Oct 05 2020
       to
       
    </div>
  </div>
  <div
    className="text-gray-texts mt-4 border border-gray-borders card"
  >
    <div
      className=" text-white font-bold py-3 px-4 card-title"
    >
      Responsibilities
    </div>
    <div
      className="py-3 px-4 card-content"
    >
      <p>
        - 
        res
      </p>
    </div>
  </div>
  <div
    className="text-gray-texts mt-4 border border-gray-borders card"
  >
    <div
      className=" text-white font-bold py-3 px-4 card-title"
    >
      Accomplishments
    </div>
    <div
      className="py-3 px-4 card-content"
    >
      <p>
        - 
        res
      </p>
    </div>
  </div>
  <div
    className="text-gray-texts mt-4 border border-gray-borders card"
  >
    <div
      className=" text-white font-bold py-3 px-4 card-title"
    >
      Favorite Project
    </div>
    <div
      className="py-3 px-4 card-content"
    >
      res
    </div>
  </div>
  <div
    className="flex justify-center mt-12"
  >
    <button
      className="add-employer-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-64 rounded-sm shadow flex justify-around"
      data-testid="next-button"
      onClick={[Function]}
      type="button"
    >
      <span
        className=""
      >
        Add another employer
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </button>
  </div>
  <div
    className="flex justify-center mt-12"
  >
    <button
      className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
      data-testid="next-button"
      onClick={[Function]}
      type="button"
    >
      <span
        className=""
      >
        Next
      </span>
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.01 11H4v2h12.01v3L20 12l-3.99-4v3z"
        />
      </svg>
    </button>
  </div>
</section>
`;
